go-mode.el

```diff
@@ -616,29 +613,92 @@ current line will be returned."
                    (looking-at ".*=[[:space:]]*$")))))
-          (or
-           ;; previous line is dangling and opens indent
-           (let ((prev-line (go-previous-line-has-dangling-op-p)))
-             (when prev-line
-               (save-excursion
-                 (goto-char prev-line)
-                 (end-of-line)
-                 (go--dangling-line-opens-indent-p))))
-
-           ;; or paren is only preceded by identifier or other parens
-           (string-match-p "^[[:space:]]*[[:word:][:multibyte:]]*(*$" (buffer-substring line-begin (point)))
-
-           ;; or a preceding paren on this line opens an indent
-           (and
-            (> (point) line-begin)
-            (progn (backward-char) (go--dangling-line-opens-indent-p)))))))))
+
+            (goto-char open-paren)
+
+            (or
+             ;; previous line is dangling and opens indent
+             (let ((prev-line (go-previous-line-has-dangling-op-p)))
+               (when prev-line
+                 (save-excursion
+                   (goto-char prev-line)
+                   (end-of-line)
+                   (go--dangling-line-opens-indent-p))))
+
+             ;; or paren is only preceded by identifier or other parens
+             (string-match-p "^[[:space:]]*[[:word:][:multibyte:]]*(*$" (buffer-substring line-begin (point)))
+
+             ;; or a preceding paren on this line opens an indent
+             (and
+              (> (point) line-begin)
+              (progn (backward-char) (go--dangling-line-opens-indent-p))))))))
+
+(defun go--commas-indent-p ()
+  "Return non-nil if in a context where dangling commas indent next line."
+  (not (or
+        (go--between-parens-p)
+        (go--in-composite-literal-p)
+        (go--in-struct-definition-p))))
+
+(defun go--in-struct-definition-p ()
+  "Return non-nil if inside a struct definition."
+  (save-excursion
+    (let ((start-paren-level (go-paren-level)))
+      (and
+       ;; inside curlies
+       (go-goto-opening-parenthesis)
+       (eq (char-after) ?{)
+       (< (go-paren-level) start-paren-level)
+
+       ;; "struct" appears before opening curly
+       (backward-word)
+       (looking-at "struct[[:space:]]")))))
+
+(defun go--in-composite-literal-p ()
+  "Return non-nil if point is in a composite literal."
+  (save-excursion
+    (let ((start-paren-level (go-paren-level)))
+      (and
+       (go-goto-opening-parenthesis)
+
+       ;; opening paren-like character is curly
+       (eq (char-after) ?{)
+
+       ;; point is before the closing curly
+       (< (go-paren-level) start-paren-level)
+
+       (or
+        ;; preceded by non space (e.g. "Foo|{")
+        (not (looking-back "[[:space:]]" (1- (point))))
+
+        ;; or curly itself is in a composite literal (e.g. "Foo{|{")
+        (go--in-composite-literal-p))))))
+
+(defun go--between-parens-p ()
+  "Return non-nil if point is between '(' and ')'.
+
+The return value is the position of the opening paren."
```

Comment:
If the function's return value has meaning beyond true/false then I don't think it should be named *-p (implying its simply a predicate function).

It seems like a better name for this function would be `go--open-paren-position`

Reply:
Fixed.